---
title: "2019-10-31_atacseq_analysis"
author: "Jonathan Lee"
date: "10/31/2019"
output: html_document
---

```{r setup, include=FALSE}
#source("http://bioconductor.org/biocLite.R")
#biocLite("preprocessCore")
#biocLite()
library(preprocessCore)
#biocLite("sva")
library(sva)
#biocLite("limma")
library(limma)
#install.packages(c("wordcloud","tm"),repos="http://cran.r-project.org")
#library(wordcloud)
#library(tm)
library(stringr)
library(ggplot2)
library(ggrepel)
#biocLite("GO.db")
library(GO.db)
library(gProfileR)
library(reactome.db)
library(fgsea)
library(biomaRt)
#biocLite("org.Hs.eg.db")
library(org.Hs.eg.db)
library(dplyr)
library(reshape2)
library(pheatmap)
library(clusterProfiler)
#install.packages("rentrez")
#library(rentrez)
library(igraph)
```


# footprint profiles

```{r, eval=T}
#devtools::install_github("GreenleafLab/chromVARmotifs")
source("http://bioconductor.org/biocLite.R")
BiocManager::install("chromVAR")
BiocManager::install("motifmatchr")
BiocManager::install("BSgenome.Hsapiens.UCSC.hg38")

library(chromVAR)
library(motifmatchr)
library(BSgenome.Hsapiens.UCSC.hg38)
library(Rsamtools)

library(stringr)

peakfile <- "~/Desktop/2020-09-04_cnr/k562_nucvinput_peaks_noblacklist.nochr.bed" # FDR 1% from macs2
peaks <- readNarrowpeaks(peakfile, width=4001) # read .narrowPeak file from macs2; default is summit

peaks@seqnames@values <- factor(paste0("chr",peaks@seqnames@values))
peaks@seqnames@values <- factor(peaks@seqnames@values, levels=as.character(peaks@seqnames@values))
peaks@seqinfo@seqnames <- paste0("chr",peaks@seqinfo@seqnames)

fnames <- list.files(path="~/Desktop/2020-09-04_cnr/", pattern="120.bam", full.names=T)
peak_mat <- list()
ecoli_reads <- list()

for(i in 1:20){
  bamfile <- fnames[i]
  ecoli_reads[[i]] <- read.table(file=str_replace_all(fnames[i], "_sorted.noM.norep.120.bam", ".ecoli_reads.txt"))

  scanned <- scanBam(bamfile, 
                     param = 
                       ScanBamParam(flag = 
                                      scanBamFlag(isMinusStrand = FALSE, 
                                                  isProperPair = TRUE),
                                    what = c("rname", "pos", "isize", "strand")))[[1]]
  
  scanned_left <- GRanges(seqnames = scanned$rname[which(scanned$isize > 0)], 
                          IRanges(start = scanned$pos[which(scanned$isize > 0)], 
                                  width = scanned$isize[which(scanned$isize > 0)]), strand = "+")
  
  scanned_left@seqnames@values <- factor(paste0("chr",scanned_left@seqnames@values))
  scanned_left@seqnames@values <- factor(scanned_left@seqnames@values, levels=as.character(scanned_left@seqnames@values))
  scanned_left@seqinfo@seqnames <- paste0("chr",scanned_left@seqinfo@seqnames)
  
  peaks_overlap <- peaks[overlapsAny(peaks, scanned_left)]
  
  x <- c(scanned_left)[as.vector(matrix(seq_len(1L * length(scanned_left)), nrow = 1L, 
                                       byrow = TRUE))]
  xcov <- coverage(x)
  
  for(chr in names(xcov)){ # for chr in chromosomes
    if(chr=="chr1"){
    peak_mat[[i]] <- 
      colSums(as.matrix(head(Views(xcov[[chr]], ranges(peaks_overlap[seqnames(peaks_overlap) == chr])), -1)))
    } else if(chr %in% levels(seqnames(peaks_overlap))) {
      peak_mat[[i]] <- peak_mat[[i]] + 
      colSums(as.matrix(head(Views(xcov[[chr]], ranges(peaks_overlap[seqnames(peaks_overlap) == chr])), -1)))
    }
  }
}

peak_mat_norm <- list()
for(i in 1:20){
  bg = mean(peak_mat[[i]][seq(length(peak_mat[[i]])-50, length(peak_mat[[i]]))])
  peak_mat_norm[[i]] <- peak_mat[[i]]/bg
}


peak_mat_ecoli <- list()
for(i in 1:20){
  #bg = mean(peak_mat[[i]][seq(length(peak_mat[[i]])-50, length(peak_mat[[i]]))])
  peak_mat_ecoli[[i]] <- peak_mat[[i]]/ecoli_reads[[i]]$V1
}
```

```{r, echo=TRUE, fig.height=2, fig.width=2.5}
library(ggplot2)
sample_names <- c("ALYREF", "ERH", "TMPO", "WBP11", "WBP11-2", "CCDC86", "NOP16", "RBM9", "H3K27me3", "IgG")

for(i in 1:8){
  df.moi <- data.frame(position = seq(-2000,2000), count=peak_mat_norm[[i]], type="Rep 1", ab="test")
  df.moi <- rbind(df.moi,
                  data.frame(position = seq(-2000,2000), count=peak_mat_norm[[i+10]], type="Rep 2", ab="test"))
  df.moi <- rbind(df.moi,
                  data.frame(position = seq(-2000,2000), count=peak_mat_norm[[10]], type="Rep 1", ab="ctrl"))
  df.moi <- rbind(df.moi,
                  data.frame(position = seq(-2000,2000), count=peak_mat_norm[[20]], type="Rep 2", ab="ctrl"))
  print(ggplot(df.moi, aes(x=position, y=count, color=ab, linetype=type)) + geom_line(alpha=0.7) + xlab("Position from iDAPT-seq\nPeak Summit") + ylab(paste0("Normalized Intensity\n",sample_names[i]," CUT&RUN")) + theme_classic()+ scale_color_manual(values=c("black", "red")) + scale_linetype_manual(values=c(3,1)) + theme(legend.position="none") + ylim(c(0.2,6.5)))
}

#+ ylim(c(0.2,5))) #+ scale_color_manual(values=c("black", "red"))
```

```{r, eval=T}
peakfile <- "~/Desktop/2020-09-04_cnr/k562_nucvinput_peaks_noblacklist.nochr.bed" # FDR 1% from macs2
peaks <- readNarrowpeaks(peakfile, width=4001) # read .narrowPeak file from macs2; default is summit

peaks@seqnames@values <- factor(paste0("chr",peaks@seqnames@values))
peaks@seqnames@values <- factor(peaks@seqnames@values, levels=as.character(peaks@seqnames@values))
peaks@seqinfo@seqnames <- paste0("chr",peaks@seqinfo@seqnames)

#fnames <- list.files(path="~/Desktop/2020-09-04_cnr/", pattern="120.bam", full.names=T)
fnames <- c("ENCFF360QBV.bam","ENCFF635HIS.bam","ENCFF023NGN.bam","ENCFF178VEA.bam","ENCFF797FEY.bam","ENCFF913HVS.bam","ENCFF234NVU.bam","ENCFF339YBR.bam","ENCFF554KEG.bam","ENCFF772PJM.bam","ENCFF195WOK","ENCFF715AMR","ENCFF910IKB","ENCFF227IZS")
fnames <- paste0("~/Desktop/2020-09-04_cnr/", fnames)

peak_mat_enc <- list()

for(i in 1:10){
  bamfile <- fnames[i]

  scanned <- scanBam(bamfile, 
                     param = 
                       ScanBamParam(what = c("rname", "pos", "strand")))[[1]]
  
  scanned_left <- GRanges(seqnames = scanned$rname, 
                          IRanges(start = scanned$pos, 
                                  width = 300), strand = "+") # arbitrary 300 bp width

  peaks_overlap <- peaks[overlapsAny(peaks, scanned_left)]
  
  x <- c(scanned_left)[as.vector(matrix(seq_len(1L * length(scanned_left)), nrow = 1L, 
                                       byrow = TRUE))]
  xcov <- coverage(x)
  
  for(chr in names(xcov)){ # for chr in chromosomes
    if(chr=="chr1"){
    peak_mat_enc[[i]] <- 
      colSums(as.matrix(head(Views(xcov[[chr]], ranges(peaks_overlap[seqnames(peaks_overlap) == chr])), -1)))
    } else if(chr %in% levels(seqnames(peaks_overlap))) {
      peak_mat_enc[[i]] <- peak_mat_enc[[i]] + 
      colSums(as.matrix(head(Views(xcov[[chr]], ranges(peaks_overlap[seqnames(peaks_overlap) == chr])), -1)))
    }
  }
}

peak_mat_enc_norm <- list()
for(i in 1:10){
#for(i in 1:1){
  bg = mean(peak_mat_enc[[i]][seq(length(peak_mat_enc[[i]])-50, length(peak_mat_enc[[i]]))])
  peak_mat_enc_norm[[i]] <- peak_mat_enc[[i]]/bg
}
```

```{r, echo=TRUE, fig.height=2, fig.width=2.5}
df.moi <- data.frame(position = seq(-2000,2000), count=peak_mat_norm[[1]], type="Rep 1", ab="test")
df.moi <- rbind(df.moi,
                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[2]], type="Rep 2", ab="test"))
df.moi <- rbind(df.moi,
                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[3]], type="Rep 2", ab="ctrl"))
#df.moi <- rbind(df.moi,
#                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[20]], type="Rep 2", ab="ctrl"))
  print(ggplot(df.moi, aes(x=position, y=count, color=ab, linetype=type)) + geom_line(alpha=0.7) + xlab("Position from iDAPT-seq\nPeak Summit") + ylab(paste0("Normalized Intensity\nMAX ChIP-seq 1")) + theme_classic()  + geom_hline(yintercept=1, lty=3)+ scale_color_manual(values=c("black", "red")) + scale_linetype_manual(values=c(3,1)) + theme(legend.position="none") + ylim(c(0.8,4)))
```
```{r, echo=TRUE, fig.height=2, fig.width=2.5}
df.moi <- data.frame(position = seq(-2000,2000), count=peak_mat_norm[[4]], type="Rep 2", ab="test")
df.moi <- rbind(df.moi,
                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[5]], type="Rep 1", ab="test"))
df.moi <- rbind(df.moi,
                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[6]], type="Rep 1", ab="ctrl"))
df.moi <- rbind(df.moi,
                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[7]], type="Rep 2", ab="ctrl"))
  print(ggplot(df.moi, aes(x=position, y=count, color=ab, linetype=type)) + geom_line(alpha=0.7) + xlab("Position from iDAPT-seq\nPeak Summit") + ylab(paste0("Normalized Intensity\nMAX ChIP-seq 2")) + theme_classic()  + geom_hline(yintercept=1, lty=3)+ scale_color_manual(values=c("black", "red")) + scale_linetype_manual(values=c(3,1)) + theme(legend.position="none") + ylim(c(0.8,4)))
```

```{r, echo=TRUE, fig.height=2, fig.width=2.5}
df.moi <- data.frame(position = seq(-2000,2000), count=peak_mat_norm[[8]], type="Rep 1", ab="test")
df.moi <- rbind(df.moi,
                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[9]], type="Rep 2", ab="test"))
df.moi <- rbind(df.moi,
                data.frame(position = seq(-2000,2000), count=peak_mat_norm[[10]], type="Rep 2", ab="ctrl"))
#df.moi <- rbind(df.moi,
#               data.frame(position = seq(-2000,2000), count=peak_mat_norm[[7]], type="Rep 2", ab="ctrl"))
  print(ggplot(df.moi, aes(x=position, y=count, color=ab, linetype=type)) + geom_line(alpha=0.7) + xlab("Position from iDAPT-seq\nPeak Summit") + ylab(paste0("Normalized Intensity\nBCLAF1 ChIP-seq")) + theme_classic()  + geom_hline(yintercept=1, lty=3)+ scale_color_manual(values=c("black", "red")) + scale_linetype_manual(values=c(3,1)) + theme(legend.position="none") + ylim(c(0.2,3)))
```

```{r, eval=T}
peakfile <- "~/Desktop/2020-09-04_cnr/k562_nucvinput_peaks_noblacklist.nochr.bed" # FDR 1% from macs2
peaks <- readNarrowpeaks(peakfile, width=4001) # read .narrowPeak file from macs2; default is summit

peaks@seqnames@values <- factor(paste0("chr",peaks@seqnames@values))
peaks@seqnames@values <- factor(peaks@seqnames@values, levels=as.character(peaks@seqnames@values))
peaks@seqinfo@seqnames <- paste0("chr",peaks@seqinfo@seqnames)

#fnames <- list.files(path="~/Desktop/2020-09-04_cnr/", pattern="120.bam", full.names=T)
fnames <- c("ENCFF360QBV.bam","ENCFF635HIS.bam","ENCFF023NGN.bam","ENCFF178VEA.bam","ENCFF797FEY.bam","ENCFF913HVS.bam","ENCFF234NVU.bam","ENCFF339YBR.bam","ENCFF554KEG.bam","ENCFF772PJM.bam","ENCFF195WOK","ENCFF715AMR","ENCFF910IKB","ENCFF227IZS")
fnames <- paste0("~/Desktop/2020-09-04_cnr/", fnames)

peak_mat_enc <- list()

for(i in 11:14){
  bamfile <- fnames[i]

  scanned <- scanBam(bamfile, 
                     param = 
                       ScanBamParam(what = c("rname", "pos", "strand")))[[1]]
  
  scanned_left <- GRanges(seqnames = scanned$rname, 
                          IRanges(start = scanned$pos, 
                                  width = 300), strand = "+") # arbitrary 200 bp width

  peaks_overlap <- peaks[overlapsAny(peaks, scanned_left)]
  
  x <- c(scanned_left)[as.vector(matrix(seq_len(1L * length(scanned_left)), nrow = 1L, 
                                       byrow = TRUE))]
  xcov <- coverage(x)
  
  for(chr in names(xcov)){ # for chr in chromosomes
    if(chr=="chr1"){
    peak_mat_enc[[i]] <- 
      colSums(as.matrix(head(Views(xcov[[chr]], ranges(peaks_overlap[seqnames(peaks_overlap) == chr])), -1)))
    } else if(chr %in% levels(seqnames(peaks_overlap))) {
      peak_mat_enc[[i]] <- peak_mat_enc[[i]] + 
      colSums(as.matrix(head(Views(xcov[[chr]], ranges(peaks_overlap[seqnames(peaks_overlap) == chr])), -1)))
    }
  }
}

#peak_mat_enc_norm <- list()
for(i in 11:14){
#for(i in 1:1){
  bg = mean(peak_mat_enc[[i]][seq(length(peak_mat_enc[[i]])-50, length(peak_mat_enc[[i]]))])
  peak_mat_enc_norm[[i]] <- peak_mat_enc[[i]]/bg
}
```

